product_abc_classification/models/abc_classification_level.py

```diff
@@ -8,28 +8,71 @@
 class ABCClassificationProfileLevel(models.Model):
     _name = "abc.classification.profile.level"
     _description = "ABC Classification Profile Level"
-    _order = "percentage desc, id desc"
+    _order = "sequence desc"
 
-    percentage = fields.Float(default=0.0, required=True, string="%")
+    classification_type = fields.Selection(
+        related="profile_id.classification_type", string="Classification Type"
+    )
+    percentage = fields.Float(default=0.0, string="%")
+    fixed = fields.Monetary(default=0.0, string="Fixed Value")
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+        string="Company",
+        related="profile_id.company_id",
+        store=True,
+    )
+    currency_id = fields.Many2one(
+        "res.currency", string="Currency", store=True, related="company_id.currency_id"
+    )
     profile_id = fields.Many2one("abc.classification.profile")
+    sequence = fields.Integer(
+        string="Sequence", compute="_compute_sequence", store=True
+    )
 
     def name_get(self):
         def _get_sort_key_percentage(rec):
             return rec.percentage
 
+        def _get_sort_key_fixed_price(rec):
+            return rec.fixed
+
+        method_dict = {
```

Comment:
You don't really need a method dict. You can take this approach instead:

https://github.com/oca/ocb/blob/e57f63a1025da93df538f3bc80c5ccb6f148735a/addons/delivery/models/delivery_carrier.py#L158-L159